gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/GemFireUpdateActivation.java

```diff
@@ -79,6 +75,16 @@ protected void executeWithResultSet(AbstractGemFireResultSet rs)
     // Asif: Obtain the Primary Key. If it is dynamic then we need to compute
     // now If it is static use it as it is
     Object pk = this.qInfo.getPrimaryKey();
+    Object[][] otherKeys = (Object[][])this.qInfo.getOtherKeys();
+
+    String whereStatement = ((UpdateQueryInfo)this.qInfo).statementSQLText;
+    int whereIndex = whereStatement.indexOf("where");
+    if (whereIndex == 0) {
+      whereIndex = whereStatement.indexOf("WHERE");
+    }
```

Comment:
I think we should avoid this hard coding later. Atleast ignoreCase should have been used. Right now a query string like update TABLEWHERE set x = 'where' where col = 'where' will pose different levels of problems. Instead we should use parsed WHERE.token.position to do a substring.